File: LeetCode/shortestWordDis1.java
public class Solution {
    public int shortestDistance(String[] words, String word1, String word2) {
        int w1=-1,w2=-1;
        int minDiff = Integer.MAX_VALUE;
        boolean flag1=false,flag2=false;
        for(int i=0;i<words.length;i++)
        {
            String currWord = words[i];
            if(currWord.equals(word1))
               {
                   flag1 = true;
                   w1 = i;
               }
            if(currWord.equals(word2))
            {
                flag2 = true;
                w2 = i;
            }
            if(flag1 && flag2)
                minDiff = Math.min(minDiff,Math.abs(w1-w2));
            
                
        }
        return minDiff;
        
    }
}

File: LeetCode/pythonSols/String/shortestWordDis1.py
class Solution(object):
    def shortestDistance(self, words, word1, word2):
        """
        :type words: List[str]
        :type word1: str
        :type word2: str
        :rtype: int
        """
        i,j=-1,-1
        sd = 1<<31
        for idx,word in enumerate(words):
            if word==word1:
                i=idx
            if word==word2:
                j = idx

            if i!=-1 and j!=-1:
                sd = min(sd,abs(i-j))

        return sd
